test/units/cli/test_galaxy.py

```diff
@@ -444,3 +460,475 @@ def test_template_ignore_similar_folder(self):
+
+        assert len(files['files']) == 5
+        assert files['format'] == 1
+        assert len(files.keys()) == 2
+
+        valid_files_entries = ['.', 'roles', 'docs', 'plugins', 'plugins/README.md']
+        for file_entry in files['files']:
+            assert file_entry['name'] in valid_files_entries
+            assert file_entry['format'] == 1
+
+            if file_entry['name'] == 'plugins/README.md':
+                assert file_entry['ftype'] == 'file'
+                assert file_entry['chksum_type'] == 'sha256'
+                assert file_entry['chksum_sha256'] == '5be7ec7b71096d56e1cc48311b6a2266b77b5fdb9d1985b5bc625787b1e857c5'
+            else:
+                assert file_entry['ftype'] == 'dir'
+                assert file_entry['chksum_type'] is None
+                assert file_entry['chksum_sha256'] is None
+
+            assert len(file_entry.keys()) == 5
+
+
+@pytest.fixture()
+def collection_install(reset_cli_args, tmp_path_factory, monkeypatch):
+    mock_install = MagicMock()
+    monkeypatch.setattr(ansible.cli.galaxy, 'install_collections', mock_install)
+
+    mock_warning = MagicMock()
+    monkeypatch.setattr(ansible.utils.display.Display, 'warning', mock_warning)
+
+    output_dir = to_text((tmp_path_factory.mktemp('test-ÅÑŚÌβŁÈ Output')))
+    yield mock_install, mock_warning, output_dir
+
+
+def test_collection_install_with_names(collection_install):
+    mock_install, mock_warning, output_dir = collection_install
+
+    galaxy_args = ['ansible-galaxy', 'collection', 'install', 'namespace.collection', 'namespace2.collection:1.0.1',
+                   '--collections-path', output_dir]
+    GalaxyCLI(args=galaxy_args).run()
+
+    collection_path = os.path.join(output_dir, 'ansible_collections')
+    assert os.path.isdir(collection_path)
+
+    assert mock_warning.call_count == 1
+    assert "The specified collections path '%s' is not part of the configured Ansible collections path" % output_dir \
+        in mock_warning.call_args[0][0]
+
+    assert mock_install.call_count == 1
+    assert mock_install.call_args[0][0] == [('namespace.collection', '*', None),
+                                            ('namespace2.collection', '1.0.1', None)]
+    assert mock_install.call_args[0][1] == collection_path
+    assert mock_install.call_args[0][2] == ['https://galaxy.ansible.com']
+    assert mock_install.call_args[0][3] is True
+    assert mock_install.call_args[0][4] is False
+    assert mock_install.call_args[0][5] is False
+    assert mock_install.call_args[0][6] is False
+    assert mock_install.call_args[0][7] is False
+
+
+def test_collection_install_with_requirements_file(collection_install):
+    mock_install, mock_warning, output_dir = collection_install
+
+    requirements_file = os.path.join(output_dir, 'requirements.yml')
+    with open(requirements_file, 'wb') as req_obj:
+        req_obj.write(b'---\ncollections:\n- namespace.coll\n- name: namespace2.coll\n  version: \'>2.0.1\'\n')
```

Comment:
It'd be more obvious that it's a YAML document if you multiline this
```suggestion
        req_obj.write(
	        b'---\n'
	        b'collections:\n'
	        b'- namespace.coll\n'
	        b'- name: namespace2.coll\n'
	        b"  version: '>2.0.1'\n"
        )
```
(or use a tripple-quoted multiline string)